backend/players/models.py

```diff
@@ -2,19 +2,110 @@
-    display_name = models.CharField(max_length=30, unique=True)
-    avatar = models.ImageField(upload_to='avatars/', default='fallback.png', blank=True)
+    user = models.OneToOneField(User, on_delete=models.CASCADE) # Registration information
+    display_name = models.CharField(max_length=30, unique=True) # Name that would be displayed to other players
+    avatar = models.ImageField(upload_to='avatars/', default='fallback.png', blank=True) # Profile picture
+    created_at = models.DateTimeField(default=timezone.now) # Date of account creation
+    updated_at = models.DateTimeField(auto_now=True) # Date of last account update
 
     def __str__(self):
-        return self.display_name
+        return f"Player: {self.display_name}"
 
 # Use signal to remove corresponding user on player deletion
 @receiver(post_delete, sender=Player)
 def delete_user_with_player(sender, instance, **kwargs):
     # Delete the associated user when the player is deleted
     if instance.user:
         instance.user.delete()
+
+
+class Tournament(models.Model):
+    name = models.CharField(max_length=100) # Name of the tournament
+    status = models.CharField(max_length=20, choices=[
+        ('scheduled', 'Scheduled'),
+        ('in_progress', 'In Progress'),
+        ('completed', 'Completed')
+    ], default='scheduled') # Status of the tournament
+    winner = models.ForeignKey(Player,
+                               on_delete=models.SET('Unknown'),
+                               null=True,
+                               blank=True,
+                               related_name='won_tournaments') # Winner of the tournament
+    created_at = models.DateTimeField(auto_now_add=True) # Date of tournament creation
+
+    def __str__(self):
+        return f"Tournament: {self.name} - Status: {self.status}"
+
+# Association table for players and tournaments
+class TournamentParticipant(models.Model):
+    tournament = models.ForeignKey(Tournament,
+                                   on_delete=models.CASCADE,
+                                   related_name='participants') # Tournament the player is participating in
+    player = models.ForeignKey(Player, on_delete=models.CASCADE) # Player participating in the tournament
+    order = models.IntegerField() # Order of the player in the tournament (For seeding purposes)
+
+    class Meta:
+        unique_together = ('tournament', 'player')
+
+    def __str__(self):
+        return f"{self.player.display_name} in {self.tournament.name}"
+
+
+class Match(models.Model):
+    tournament = models.ForeignKey(Tournament,
+                                   on_delete=models.CASCADE,
+                                   related_name='matches',
+                                   null=True,
+                                   blank=True) # Tournament the match is part of
+    player1 = models.ForeignKey(Player,
+                                on_delete=models.CASCADE,
```

Comment:
Consider keeping matches even if one of the player is removed. It will be still helpful to keep the stats for the second player. Or even if both players are removed. We can keep track of amount of matches played.

```suggestion
                                on_delete=models.SET_NULL,
```